logs/field.go

```diff
@@ -0,0 +1,211 @@
+package logs
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/trace"
+)
+
+// Field represents typed log field (a key-value pair). Adapters should determine
+// Field's type based on Type and use the corresponding getter method to retrieve
+// the value:
+//
+//	switch f.Type {
+//	case logs.IntType:
+//	    var i int = f.Int()
+//	    // handle int value
+//	case logs.StringType:
+//	    var s string = f.String()
+//	    // handle string value
+//	//...
+//	}
+//
+// Getter methods must not be called on fields with wrong Type (e.g. calling String()
+// on fields with Type != StringType).
+type Field struct {
+	Type FieldType
+	Key  string
+
+	vint int64
+	vstr string
+	vany interface{}
+}
+
+func String(key string, value string) Field {
+	return Field{
+		Type: StringType,
+		Key:  key,
+		vstr: value,
+	}
+}
+
+func (f Field) String() string {
+	return f.vstr
+}
+
+func Int(key string, value int) Field {
+	return Field{
+		Type: IntType,
+		Key:  key,
+		vint: int64(value),
+	}
+}
+
+func (f Field) Int() int {
+	return int(f.vint)
+}
+
+func Bool(key string, value bool) Field {
+	var vint int64
+	if value {
+		vint = 1
+	} else {
+		vint = 0
+	}
+	return Field{
+		Type: BoolType,
+		Key:  key,
+		vint: vint,
+	}
+}
+
+func (f Field) Bool() bool {
+	return f.vint != 0
+}
+
+func Duration(key string, value time.Duration) Field {
+	return Field{
+		Type: DurationType,
+		Key:  key,
+		vint: value.Nanoseconds(),
+	}
+}
+
+func (f Field) Duration() time.Duration {
+	return time.Nanosecond * time.Duration(f.vint)
+}
+
+func Strings(key string, value []string) Field {
+	return Field{
+		Type: StringsType,
+		Key:  key,
+		vany: value,
+	}
+}
+
+func (f Field) Strings() []string {
+	return f.vany.([]string)
+}
+
+// NamedError adds error field. If value is nil, resulting Field will be
+// of NilType instead of ErrorType.
+func NamedError(key string, value error) Field {
+	if value == nil {
+		return nilField(key)
+	}
+	return Field{
+		Type: ErrorType,
+		Key:  key,
+		vany: value,
+	}
+}
+
+// Error is the same as NamedError("error", value)
+func Error(value error) Field {
+	return NamedError("error", value)
+}
+
+func (f Field) Error() error {
+	return f.vany.(error)
```

Comment:
This line will panic `if f.vany == nil`. To guard against it I specified in the comment that getter `X` **must not** be called if `f.Type != XType`. Need to verify if this is ok behavior for SDK.